velox/py/plan_builder/PyPlanBuilder.h

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/core/PlanNode.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+#include "velox/py/type/PyType.h"
+
+namespace facebook::velox::py {
+
+/// Called from when the Python module is loaded/initialized to register all
+/// resources and structures needed by Velox.
+void registerAllResources();
+
+class PyVector;
+
+/// Wrapper class to expose Velox plan nodes to Python.
+class PyPlanNode {
+ public:
+  explicit PyPlanNode(core::PlanNodePtr planNode = nullptr)
+      : planNode_(std::move(planNode)) {}
+
+  core::PlanNodePtr getPlanNode() const {
+    return planNode_;
+  }
+
+  std::string id() const {
+    return std::string{planNode_->id()};
+  }
+
+  std::string serialize() const {
+    return folly::toJson(planNode_->serialize());
+  }
+
+  std::string name() const {
+    if (!planNode_) {
```

Comment:
What is the use case for planNode_ being null?

Reply:
In theory one can instantiate an empty plan builder and get the plan node at its nullptr head.  But now that I think about it, it's probably better to throw and assume it's always non-null.